components/family-studies/migrations/src/main/java/org/phenotips/studies/family/migrations/PhenotipsFamilyMigrations.java

```diff
@@ -176,17 +181,78 @@ public JSONObject processPedigree(JSONObject data, String patientId)
                 continue;
             }
 
-            JSONObject properties = (JSONObject) node.get("prop");
+            JSONObject properties = node.optJSONObject("prop");
+            if (properties == null) {
+                properties = new JSONObject();
+                node.accumulate("prop", properties);
+            }
             properties.accumulate("phenotipsId", patientId);
             break;
         }
 
-        data.accumulate("probandNodeID", 0);
-        data.accumulate("JSON_version", "1.0");
+        data.put("probandNodeID", 0);
+        data.put("JSON_version", "1.0");
         return data;
     }
 
     /**
+     * Finds a pedigree node linked to patientID and adds the given comment to it's free-text comment.
+     *
+     * @param pedigree pedigree as JSON
+     * @param patientId the PhenoTips patient id
+     * @param commentAddition text of comment addition
+     */
+    public void updatePedigreeComment(JSONObject pedigree, String patientId, String commentAddition)
+    {
+        JSONArray gg = pedigree.getJSONArray("GG");
+        for (int pedigreeId = 0; pedigreeId < gg.length(); pedigreeId++) {
+            JSONObject node = gg.getJSONObject(pedigreeId);
+
+            JSONObject properties = node.optJSONObject("prop");
+            if (properties == null) {
```

Comment:
as above, there is no difference for us if legth is 0 or not 0, even if it is not 0 it may not contain what we are looking for, and that is taken care for below. I'm not sure JSONObject even has length() (unlike JSONArray)